aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -128,9 +113,11 @@ void CUDAGraph::capture_begin(MempoolId_t pool/*=0*/, cudaStreamCaptureMode capt
     TORCH_INTERNAL_ASSERT(!(pool.first && pool.second));
     mempool_id_ = pool;
   } else {
-    // User did not ask us to share a mempool. Use our own id_ as our mempool_id_.
+    // User did not ask us to share a mempool. Create graph pool handle using is_user_created=false.
     // Sets just the first value, to distinguish it from MempoolId_ts created by graph_pool_handle().
-    mempool_id_ = {id_, 0};
+    auto mempool = c10::cuda::MemPool({}, false);
+    mempool_id_ = mempool.id();
```

Comment:
So is cuda graphs the only thing that actually uses this right now? In particular, how would I allocate into the mempool from a non-cudagraph use case?

Reply:
That's coming up in the next PR. I'm breaking this https://github.com/pytorch/pytorch/pull/125722 into multiple PRs, and wanted a separate review on CUDACachingAllocator changes. If you look at the changes in the CUDACachingAllocator there, you can see how the caching allocator can query if there is an active mempool, and then based on that route the allocations.